packages/components/src/range-control/index.tsx

```diff
@@ -166,13 +181,16 @@ function UnforwardedRangeControl(
 	};
 
 	const handleOnReset = () => {
-		let resetValue: number | null = parseFloat( `${ resetFallbackValue }` );
-		let onChangeResetValue: number | undefined = resetValue;
+		// Reset to `resetFallbackValue` if defined, otherwise set internal value
+		// to `null`.
+		const resetValue = Number.isNaN( resetFallbackValue )
+			? null
+			: resetFallbackValue ?? null;
 
-		if ( isNaN( resetValue ) ) {
-			resetValue = null;
-			onChangeResetValue = undefined;
-		}
+		// const resetValue = computeResetValue( {
+		// 	resetFallbackValue,
+		// 	initialPosition,
+		// } );
```

Comment:
@mirka pinging you for some early feedback.

I took a look at the component, specifically about how it behaves when clicking the reset button. This is what I gathered:

- when `resetFallbackValue` is provided:
   - we call `setValue(resetFallbackValue)`, but if the component is controlled, the external `value` still prevails;
   - onChange() is called with `resetFallbackValue`, which causes the parent component to update the value prop, which causes the internal value to update to resetFallbackValue;
- when `initialPosition` is provided:
   - we call `setValue(null)`, but if the component is controlled, the external `value` still prevails;
   - onChange() is called with `undefined`, which causes the parent component to update the value prop, which causes the internal value to update to fallback to `initialPosition`;
- when neither resetFallbackValue nor initialPosition are provided:
   - we call `setValue(null)`, but if the component is controlled, the external `value` still prevails;
   - onChange() is called with `undefined`, which causes the parent component to update the value prop, but since there is not `initialPosition` to fallback to, the internal value stays `null`;

I extracted the `computeResetValue` function which kind of codifies the expected reset value logic. But if I applied it to the `handleOnReset` function, it would cause some changes to how the component works (especially to the arguments of `onChange`).

My gut feeling is to keep this PR's scope more limited and focus on disabling the `reset` button correctly.

Separately, IMO we should consider re-thinking the controlled/uncontrolled logic entirely, so that `onChange` and the reset functionality behave in a more "standard" fashion (from a quick test, the reset button doesn't seem to work at all in uncontrolled mode). If we think that the changes are breaking enough, we may even want to consider an ariakit rewrite in the process (and consider the plan highlighted in #40507)

Reply:
Great investigation, thank you. It's a good look into the kind of things we'll be dealing with for #57004.

> My gut feeling is to keep this PR's scope more limited and focus on disabling the `reset` button correctly.

Definitely. And also, I just noticed that `allowReset` may be a candidate for deprecation. It doesn't look like it's used anywhere in the codebase anymore, as well as the one on BoxControl. Makes sense, because we've delegated all reset functionality to the ToolsPanel.

Reply:
Cool, i'll keep the scope of this PR limited to fixing the reset button disabled behavior.

> I just noticed that allowReset may be a candidate for deprecation.

True, we could also open a follow-up PR to deprecated `allowReset` and `resetFallbackValue`, although it's probably not a high priority.